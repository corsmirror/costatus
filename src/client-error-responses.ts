/**
 * 400 Bad Request
 *
 * The server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).
 *
 * ```ts
 * import { BAD_REQUEST } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/400}
 */
export const BAD_REQUEST = 400;

/**
 * 401 Unauthorized
 *
 * Although the HTTP standard specifies "unauthorized", semantically this response means "unauthenticated". That is, the client must authenticate itself to get the requested response.
 *
 * ```ts
 * import { UNAUTHORIZED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/401}
 */
export const UNAUTHORIZED = 401;

/**
 * 402 Payment Required
 *
 * This response code is reserved for future use. The initial aim for creating this code was using it for digital payment systems, however this status code is used very rarely and no standard convention exists.
 *
 * ```ts
 * import { PAYMENT_REQUIRED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/402}
 */
export const PAYMENT_REQUIRED = 402;

/**
 * 403 Forbidden
 *
 * The client does not have access rights to the content; that is, it is unauthorized, so the server is refusing to give the requested resource. Unlike `401 Unauthorized`, the client's identity is known to the server.
 *
 * ```ts
 * import { FORBIDDEN } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/403}
 */
export const FORBIDDEN = 403;

/**
 * 404 Not Found
 *
 * The server cannot find the requested resource. In the browser, this means the URL is not recognized. In an API, this can also mean that the endpoint is valid but the resource itself does not exist. Servers may also send this response instead of `403 Forbidden` to hide the existence of a resource from an unauthorized client. This response code is probably the most well known due to its frequent occurrence on the web.
 *
 * ```ts
 * import { NOT_FOUND } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/404}
 */
export const NOT_FOUND = 404;

/**
 * 405 Method Not Allowed
 *
 * The request method is known by the server but is not supported by the target resource. For example, an API may not allow calling `DELETE` to remove a resource.
 *
 * ```ts
 * import { METHOD_NOT_ALLOWED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/405}
 */
export const METHOD_NOT_ALLOWED = 405;

/**
 * 406 Not Acceptable
 *
 * This response is sent when the web server, after performing server-driven content negotiation, doesn't find any content that conforms to the criteria given by the user agent.
 *
 * ```ts
 * import { NOT_ACCEPTABLE } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/406}
 */
export const NOT_ACCEPTABLE = 406;

/**
 * 407 Proxy Authentication Required
 *
 * This is similar to `401 Unauthorized` but authentication is needed to be done by a proxy.
 *
 * ```ts
 * import { PROXY_AUTHENTICATION_REQUIRED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/407}
 */
export const PROXY_AUTHENTICATION_REQUIRED = 407;

/**
 * 408 Request Timeout
 *
 * This response is sent on an idle connection by some servers, even without any previous request by the client. It means that the server would like to shut down this unused connection. This response is used much more since some browsers, like Chrome, Firefox 27+, or IE9, use HTTP pre-connection mechanisms to speed up surfing. Also note that some servers merely shut down the connection without sending this message.
 *
 * ```ts
 * import { REQUEST_TIMEOUT } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/408}
 */
export const REQUEST_TIMEOUT = 408;

/**
 * 409 Conflict
 *
 * This response is sent when a request conflicts with the current state of the server.
 *
 * ```ts
 * import { CONFLICT } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/409}
 */
export const CONFLICT = 409;

/**
 * 410 Gone
 *
 * This response is sent when the requested content has been permanently deleted from server, with no forwarding address. Clients are expected to remove their caches and links to the resource. The HTTP specification intends this status code to be used for "limited-time, promotional services". APIs should not feel compelled to indicate resources that have been deleted with this status code.
 *
 * ```ts
 * import { GONE } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/410}
 */
export const GONE = 410;

/**
 * 411 Length Required
 *
 * Server rejected the request because the `Content-Length` header field is not defined and the server requires it.
 *
 * ```ts
 * import { LENGTH_REQUIRED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/411}
 */
export const LENGTH_REQUIRED = 411;

/**
 * 412 Precondition Failed
 *
 * The client has indicated preconditions in its headers which the server does not meet.
 *
 * ```ts
 * import { PRECONDITION_FAILED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/412}
 */
export const PRECONDITION_FAILED = 412;

/**
 * 413 Payload Too Large
 *
 * Request entity is larger than limits defined by server. The server might close the connection or return an `Retry-After` header field.
 *
 * ```ts
 * import { PAYLOAD_TOO_LARGE } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/413}
 */
export const PAYLOAD_TOO_LARGE = 413;

/**
 * 414 URI Too Long
 *
 * The URI requested by the client is longer than the server is willing to interpret.
 *
 * ```ts
 * import { URI_TOO_LONG } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/414}
 */
export const URI_TOO_LONG = 414;

/**
 * 415 Unsupported Media Type
 *
 * The media format of the requested data is not supported by the server, so the server is rejecting the request.
 *
 * ```ts
 * import { UNSUPPORTED_MEDIA_TYPE } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/415}
 */
export const UNSUPPORTED_MEDIA_TYPE = 415;

/**
 * 416 Range Not Satisfiable
 *
 * The range specified by the `Range` header field in the request cannot be fulfilled. It's possible that the range is outside the size of the target URI's data.
 *
 * ```ts
 * import { RANGE_NOT_SATISFIABLE } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/416}
 */
export const RANGE_NOT_SATISFIABLE = 416;

/**
 * 417 Expectation Failed
 *
 * This response code means the expectation indicated by the `Expect` request header field cannot be met by the server.
 *
 * ```ts
 * import { EXPECTATION_FAILED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/417}
 */
export const EXPECTATION_FAILED = 417;

/**
 * 418 I'm a teapot
 *
 * The server refuses the attempt to brew coffee with a teapot.
 *
 * ```ts
 * import { IM_A_TEAPOT } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/418}
 */
export const IM_A_TEAPOT = 418;

/**
 * 421 Misdirected Request
 *
 * The request was directed at a server that is not able to produce a response. This can be sent by a server that is not configured to produce responses for the combination of scheme and authority that are included in the request URI.
 *
 * ```ts
 * import { MISDIRECTED_REQUEST } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/421}
 */
export const MISDIRECTED_REQUEST = 421;

/**
 * 422 Unprocessable Content
 *
 * The request was well-formed but was unable to be followed due to semantic errors.
 *
 * ```ts
 * import { UNPROCESSABLE_CONTENT } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/422}
 */
export const UNPROCESSABLE_CONTENT = 422;

/**
 * 423 Locked
 *
 * The resource that is being accessed is locked.
 *
 * ```ts
 * import { LOCKED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/423}
 */
export const LOCKED = 423;

/**
 * 424 Failed Dependency
 *
 * The request failed due to failure of a previous request.
 *
 * ```ts
 * import { FAILED_DEPENDENCY } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/424}
 */
export const FAILED_DEPENDENCY = 424;

/**
 * 425 Too Early
 *
 * Indicates that the server is unwilling to risk processing a request that might be replayed.
 *
 * ```ts
 * import { TOO_EARLY } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/425}
 */
export const TOO_EARLY = 425;

/**
 * 426 Upgrade Required
 *
 * The server refuses to perform the request using the current protocol but might be willing to do so after the client upgrades to a different protocol. The server sends an `Upgrade` header in a 426 response to indicate the required protocol(s).
 *
 * ```ts
 * import { UPGRADE_REQUIRED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/426}
 */
export const UPGRADE_REQUIRED = 426;

/**
 * 428 Precondition Required
 *
 * The origin server requires the request to be conditional. This response is intended to prevent the 'lost update' problem, where a client `GET`s a resource's state, modifies it and `PUT`s it back to the server, when meanwhile a third party has modified the state on the server, leading to a conflict.
 *
 * ```ts
 * import { PRECONDITION_REQUIRED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/428}
 */
export const PRECONDITION_REQUIRED = 428;

/**
 * 429 Too Many Requests
 *
 * The user has sent too many requests in a given amount of time ("rate limiting").
 *
 * ```ts
 * import { TOO_MANY_REQUESTS } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/429}
 */
export const TOO_MANY_REQUESTS = 429;

/**
 * 431 Request Header Fields Too Large
 *
 * The server is unwilling to process the request because its header fields are too large. The request may be resubmitted after reducing the size of the request header fields.
 *
 * ```ts
 * import { REQUEST_HEADER_FIELDS_TOO_LARGE } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/431}
 */
export const REQUEST_HEADER_FIELDS_TOO_LARGE = 431;

/**
 * 451 Unavailable For Legal Reasons
 *
 * The user agent requested a resource that cannot legally be provided, such as a web page censored by a government.
 *
 * ```ts
 * import { UNAVAILABLE_FOR_LEGAL_REASONS } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/451}
 */
export const UNAVAILABLE_FOR_LEGAL_REASONS = 451;
