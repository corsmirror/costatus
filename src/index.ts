/**
 * HTTP response status codes
 *
 * 1. Informational responses (`100` - `199`)
 * 2. Successful responses (`200` – `299`)
 * 3. Redirection messages (`300` – `399`)
 * 4. Client error responses (`400` – `499`)
 * 5. Server error responses (`500` – `599`)
 *
 * @see {@link https://developer.mozilla.org/docs/Web/HTTP/Status}
 */

export * from './client-error-responses';
export * from './information-responses';
export * from './redirection-messages';
export * from './server-error-responses';
export * from './successful-responses';
