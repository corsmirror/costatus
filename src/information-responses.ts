/**
 * 100 Continue
 *
 * This interim response indicates that the client should continue the request or ignore the response if the request is already finished.
 *
 * ```ts
 * import { CONTINUE } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/100}
 */
export const CONTINUE = 100;

/**
 * 101 Switching Protocols
 *
 * This code is sent in response to an `Upgrade` request header from the client and indicates the protocol the server is switching to.
 *
 * ```ts
 * import { SWITCHING_PROTOCOLS } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/101}
 */
export const SWITCHING_PROTOCOLS = 101;

/**
 * 102 Processing
 *
 * This code indicates that the server has received and is processing the request, but no response is available yet.
 *
 * ```ts
 * import { PROCESSING } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/102}
 */
export const PROCESSING = 102;

/**
 * 103 Early Hints
 *
 * This status code is primarily intended to be used with the `Link` header, letting the user agent start preloading resources while the server prepares a response.
 *
 * ```ts
 * import { EARLY_HINTS } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/103}
 */
export const EARLY_HINTS = 103;
