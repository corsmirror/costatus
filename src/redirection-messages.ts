/**
 * 300 Multiple Choices
 *
 * The request has more than one possible response. The user agent or user should choose one of them. (There is no standardized way of choosing one of the responses, but HTML links to the possibilities are recommended so the user can pick.)
 *
 * ```ts
 * import { MULTIPLE_CHOICES } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/300}
 */
export const MULTIPLE_CHOICES = 300;

/**
 * 301 Moved Permanently
 *
 * The URL of the requested resource has been changed permanently. The new URL is given in the response.
 *
 * ```ts
 * import { MOVED_PERMANENTLY } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/301}
 */
export const MOVED_PERMANENTLY = 301;

/**
 * 302 Found
 *
 * This response code means that the URI of requested resource has been changed _temporarily_. Further changes in the URI might be made in the future. Therefore, this same URI should be used by the client in future requests.
 *
 * ```ts
 * import { FOUND } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/302}
 */
export const FOUND = 302;

/**
 * 303 See Other
 *
 * The server sent this response to direct the client to get the requested resource at another URI with a GET request.
 *
 * ```ts
 * import { SEE_OTHER } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/303}
 */
export const SEE_OTHER = 303;

/**
 * 304 Not Modified
 *
 * This is used for caching purposes. It tells the client that the response has not been modified, so the client can continue to use the same cached version of the response.
 *
 * ```ts
 * import { NOT_MODIFIED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/304}
 */
export const NOT_MODIFIED = 304;

/**
 * 305 Use Proxy
 *
 * @deprecated
 *
 * Defined in a previous version of the HTTP specification to indicate that a requested response must be accessed by a proxy. It has been deprecated due to security concerns regarding in-band configuration of a proxy.
 *
 * ```ts
 * import { USE_PROXY } from 'costatus';
 * ```
 */
export const USE_PROXY = 305;

/**
 * 306 unused
 *
 * @deprecated
 *
 * This response code is no longer used; it is just reserved. It was used in a previous version of the HTTP/1.1 specification.
 *
 * ```ts
 * import { UNUSED } from 'costatus';
 * ```
 */
export const UNUSED = 306;

/**
 * 307 Temporary Redirect
 *
 * The server sends this response to direct the client to get the requested resource at another URI with the same method that was used in the prior request. This has the same semantics as the `302 Found` HTTP response code, with the exception that the user agent _must not_ change the HTTP method used: if a `POST` was used in the first request, a `POST` must be used in the second request.
 *
 * ```ts
 * import { TEMPORARY_REDIRECT } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/307}
 */
export const TEMPORARY_REDIRECT = 307;

/**
 * 308 Permanent Redirect
 *
 * This means that the resource is now permanently located at another URI, specified by the `Location:` HTTP Response header. This has the same semantics as the `301 Moved Permanently` HTTP response code, with the exception that the user agent _must not_ change the HTTP method used: if a `POST` was used in the first request, a `POST` must be used in the second request.
 *
 * ```ts
 * import { PERMANENT_REDIRECT } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/308}
 */
export const PERMANENT_REDIRECT = 308;
