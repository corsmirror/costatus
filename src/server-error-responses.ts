/**
 * 500 Internal Server Error
 *
 * The server has encountered a situation it does not know how to handle.
 *
 * ```ts
 * import { INTERNAL_SERVER_ERROR } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/500}
 */
export const INTERNAL_SERVER_ERROR = 500;

/**
 * 501 Not Implemented
 *
 * The request method is not supported by the server and cannot be handled. The only methods that servers are required to support (and therefore that must not return this code) are `GET` and `HEAD`.
 *
 * ```ts
 * import { NOT_IMPLEMENTED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/501}
 */
export const NOT_IMPLEMENTED = 501;

/**
 * 502 Bad Gateway
 *
 * This error response means that the server, while working as a gateway to get a response needed to handle the request, got an invalid response.
 *
 * ```ts
 * import { BAD_GATEWAY } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/502}
 */
export const BAD_GATEWAY = 502;

/**
 * 503 Service Unavailable
 *
 * The server is not ready to handle the request. Common causes are a server that is down for maintenance or that is overloaded. Note that together with this response, a user-friendly page explaining the problem should be sent. This response should be used for temporary conditions and the `Retry-After` HTTP header should, if possible, contain the estimated time before the recovery of the service. The webmaster must also take care about the caching-related headers that are sent along with this response, as these temporary condition responses should usually not be cached.
 *
 * ```ts
 * import { SERVICE_UNAVAILABLE } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/503}
 */
export const SERVICE_UNAVAILABLE = 503;

/**
 * 504 Gateway Timeout
 *
 * This error response is given when the server is acting as a gateway and cannot get a response in time.
 *
 * ```ts
 * import { GATEWAY_TIMEOUT } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/504}
 */
export const GATEWAY_TIMEOUT = 504;

/**
 * 505 HTTP Version Not Supported
 *
 * The HTTP version used in the request is not supported by the server.
 *
 * ```ts
 * import { HTTP_VERSION_NOT_SUPPORTED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/505}
 */
export const HTTP_VERSION_NOT_SUPPORTED = 505;

/**
 * 506 Variant Also Negotiates
 *
 * The server has an internal configuration error: the chosen variant resource is configured to engage in transparent content negotiation itself, and is therefore not a proper end point in the negotiation process.
 *
 * ```ts
 * import { VARIANT_ALSO_NEGOTIATES } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/506}
 */
export const VARIANT_ALSO_NEGOTIATES = 506;

/**
 * 507 Insufficient Storage
 *
 * The method could not be performed on the resource because the server is unable to store the representation needed to successfully complete the request.
 *
 * ```ts
 * import { INSUFFICIENT_STORAGE } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/507}
 */
export const INSUFFICIENT_STORAGE = 507;

/**
 * 508 Loop Detected
 *
 * The server detected an infinite loop while processing the request.
 *
 * ```ts
 * import { LOOP_DETECTED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/508}
 */
export const LOOP_DETECTED = 508;

/**
 * 510 Not Extended
 *
 * Further extensions to the request are required for the server to fulfill it.
 *
 * ```ts
 * import { NOT_EXTENDED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/510}
 */
export const NOT_EXTENDED = 510;

/**
 * 511 Network Authentication Required
 *
 * Indicates that the client needs to authenticate to gain network access.
 *
 * ```ts
 * import { NETWORK_AUTHENTICATION_REQUIRED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/511}
 */
export const NETWORK_AUTHENTICATION_REQUIRED = 511;
