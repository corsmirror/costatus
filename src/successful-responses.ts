/**
 * 200 OK
 *
 * The request succeeded. The result meaning of "success" depends on the HTTP method:
 *
 * - `GET`: The resource has been fetched and transmitted in the message body.
 * - `HEAD`: The representation headers are included in the response without any message body.
 * - `PUT` or `POST`: The resource describing the result of the action is transmitted in the message body.
 * - `TRACE`: The message body contains the request message as received by the server.
 *
 * ```ts
 * import { OK } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/200}
 */
export const OK = 200;

/**
 * 201 Created
 *
 * The request succeeded, and a new resource was created as a result. This is typically the response sent after `POST` requests, or some `PUT` requests.
 *
 * ```ts
 * import { CREATED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/201}
 */
export const CREATED = 201;

/**
 * 202 Accepted
 *
 * The request has been received but not yet acted upon. It is noncommittal, since there is no way in HTTP to later send an asynchronous response indicating the outcome of the request. It is intended for cases where another process or server handles the request, or for batch processing.
 *
 * ```ts
 * import { ACCEPTED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/202}
 */
export const ACCEPTED = 202;

/**
 * 203 Non-Authoritative Information
 *
 * This response code means the returned metadata is not exactly the same as is available from the origin server, but is collected from a local or a third-party copy. This is mostly used for mirrors or backups of another resource. Except for that specific case, the 200 OK response is preferred to this status.
 *
 * ```ts
 * import { NON_AUTHORITATIVE_INFORMATION } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/203}
 */
export const NON_AUTHORITATIVE_INFORMATION = 203;

/**
 * 204 No Content
 *
 * There is no content to send for this request, but the headers may be useful. The user agent may update its cached headers for this resource with the new ones.
 *
 * ```ts
 * import { NO_CONTENT } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/204}
 */
export const NO_CONTENT = 204;

/**
 * 205 Reset Content
 *
 * Tells the user agent to reset the document which sent this request.
 *
 * ```ts
 * import { RESET_CONTENT } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/205}
 */
export const RESET_CONTENT = 205;

/**
 * 206 Partial Content
 *
 * This response code is used when the `Range` header is sent from the client to request only part of a resource.
 *
 * ```ts
 * import { PARTIAL_CONTENT } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/206}
 */
export const PARTIAL_CONTENT = 206;

/**
 * 207 Multi-Status
 *
 * Conveys information about multiple resources, for situations where multiple status codes might be appropriate.
 *
 * ```ts
 * import { MULTI_STATUS } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/207}
 */
export const MULTI_STATUS = 207;

/**
 * 208 Already Reported
 *
 * Used inside a `<dav:propstat>` response element to avoid repeatedly enumerating the internal members of multiple bindings to the same collection.
 *
 * ```ts
 * import { ALREADY_REPORTED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/208}
 */
export const ALREADY_REPORTED = 208;

/**
 * 226 IM Used
 *
 * The server has fulfilled a `GET` request for the resource, and the response is a representation of the result of one or more instance-manipulations applied to the current instance.
 *
 * ```ts
 * import { IM_USED } from 'costatus';
 * ```
 *
 * {@link https://developer.mozilla.org/docs/Web/HTTP/Status/226}
 */
export const IM_USED = 226;
