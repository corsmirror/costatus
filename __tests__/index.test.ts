import * as costatus from '../src';

it('exports HTTP response status codes', () => {
  expect(costatus).toMatchSnapshot();
});

it('exports 200 OK', () => {
  expect(costatus.OK).toBe(200);
});

it('exports 404 Not Found', () => {
  expect(costatus.NOT_FOUND).toBe(404);
});

it('exports 500 Internal Server Error', () => {
  expect(costatus.INTERNAL_SERVER_ERROR).toBe(500);
});

it('does not contain duplicate status codes', () => {
  const codes = Object.values(costatus);
  // https://stackoverflow.com/a/32122760
  const duplicates = codes.filter(
    (element, index, array) => array.indexOf(element) !== index,
  );
  expect(duplicates).toHaveLength(0);
});
