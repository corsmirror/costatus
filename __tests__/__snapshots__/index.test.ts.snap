// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exports HTTP response status codes 1`] = `
{
  "ACCEPTED": 202,
  "ALREADY_REPORTED": 208,
  "BAD_GATEWAY": 502,
  "BAD_REQUEST": 400,
  "CONFLICT": 409,
  "CONTINUE": 100,
  "CREATED": 201,
  "EARLY_HINTS": 103,
  "EXPECTATION_FAILED": 417,
  "FAILED_DEPENDENCY": 424,
  "FORBIDDEN": 403,
  "FOUND": 302,
  "GATEWAY_TIMEOUT": 504,
  "GONE": 410,
  "HTTP_VERSION_NOT_SUPPORTED": 505,
  "IM_A_TEAPOT": 418,
  "IM_USED": 226,
  "INSUFFICIENT_STORAGE": 507,
  "INTERNAL_SERVER_ERROR": 500,
  "LENGTH_REQUIRED": 411,
  "LOCKED": 423,
  "LOOP_DETECTED": 508,
  "METHOD_NOT_ALLOWED": 405,
  "MISDIRECTED_REQUEST": 421,
  "MOVED_PERMANENTLY": 301,
  "MULTIPLE_CHOICES": 300,
  "MULTI_STATUS": 207,
  "NETWORK_AUTHENTICATION_REQUIRED": 511,
  "NON_AUTHORITATIVE_INFORMATION": 203,
  "NOT_ACCEPTABLE": 406,
  "NOT_EXTENDED": 510,
  "NOT_FOUND": 404,
  "NOT_IMPLEMENTED": 501,
  "NOT_MODIFIED": 304,
  "NO_CONTENT": 204,
  "OK": 200,
  "PARTIAL_CONTENT": 206,
  "PAYLOAD_TOO_LARGE": 413,
  "PAYMENT_REQUIRED": 402,
  "PERMANENT_REDIRECT": 308,
  "PRECONDITION_FAILED": 412,
  "PRECONDITION_REQUIRED": 428,
  "PROCESSING": 102,
  "PROXY_AUTHENTICATION_REQUIRED": 407,
  "RANGE_NOT_SATISFIABLE": 416,
  "REQUEST_HEADER_FIELDS_TOO_LARGE": 431,
  "REQUEST_TIMEOUT": 408,
  "RESET_CONTENT": 205,
  "SEE_OTHER": 303,
  "SERVICE_UNAVAILABLE": 503,
  "SWITCHING_PROTOCOLS": 101,
  "TEMPORARY_REDIRECT": 307,
  "TOO_EARLY": 425,
  "TOO_MANY_REQUESTS": 429,
  "UNAUTHORIZED": 401,
  "UNAVAILABLE_FOR_LEGAL_REASONS": 451,
  "UNPROCESSABLE_CONTENT": 422,
  "UNSUPPORTED_MEDIA_TYPE": 415,
  "UNUSED": 306,
  "UPGRADE_REQUIRED": 426,
  "URI_TOO_LONG": 414,
  "USE_PROXY": 305,
  "VARIANT_ALSO_NEGOTIATES": 506,
}
`;
